mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -347,11 +359,14 @@ def set_image_index_range_label(self, range_min: int, range_max: int) -> None:
     def set_tof_axis_label(self, tof_axis_label: str) -> None:
         self.spectrum.setLabel('bottom', text=tof_axis_label)
 
-    def set_wavelength_range_label(self, range_min: float, range_max: float) -> None:
-        if not hasattr(self, "_wavelength_range_label"):
+    def set_unit_range_label(self,
```

Comment:
Can this be unified with `set_tof_range_label`, `set_image_index_range_label`, `set_tof_axis_label`?

Could there be one function in the presenter that gets called when the units change, creates the full label, updates the spectrum views on both tabs, updates the menus?

Reply:
while this is quite a significant change to suggest as It will require refactoring the majority of the modifications introduced in this pull request, I will try to implement it to try and centralise things